cellprofiler/modules/plugins/imagetemplate.py

```diff
@@ -1,36 +1,115 @@
-'''<b>ImageTemplate</b> - an example image processing module
-<hr>
-This is an example of a module that takes one image as an input and
-produces a second image for downstream processing. You can use this as
-a starting point for your own module: rename this file and put it in your
-plugins directory.
+# coding=utf-8
 
-The text you see here will be displayed as the help for your module. You
-can use HTML markup here and in the settings text; the Python HTML control
-does not fully support the HTML specification, so you may have to experiment
-to get it to display correctly.
-'''
 #################################
 #
 # Imports from useful Python libraries
 #
 #################################
 
-import numpy as np
-from scipy.ndimage import gaussian_gradient_magnitude, correlate1d
+import numpy
+import scipy.ndimage
 
 #################################
 #
 # Imports from CellProfiler
 #
-# The package aliases are the standard ones we use
-# throughout the code.
-#
 ##################################
 
-import cellprofiler.image as cpi
-import cellprofiler.module as cpm
-import cellprofiler.setting as cps
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+
+__doc__ = """\
+ImageTemplate
+=============
+
+**ImageTemplate** is an example image processing module. It's recommended to
+put a brief description of this module here and go into more detail below.
+
+This is an example of a module that takes one image as an input and
+produces a second image for downstream processing. You can use this as
+a starting point for your own module: rename this file and put it in your
+plugins directory.
+
+The text you see here will be displayed as the help for your module, formatted
+as `reStructuredText <http://docutils.sourceforge.net/rst.html>`_.
+
+Note whether or not this module supports 3D image data and respects masks.
+A module which respects masks applies an image's mask and operates only on
+the unmasked data. Update the table below to indicate which image processing
```

Comment:
This seems confusing to me. It applies the mask but operates on the _unmasked_ data? (I'm probably misunderstanding how this works)

Reply:
Right. A mask obstructs unwanted pixels. It's a boolean matrix where `False` represents "masked" pixels (pixels we don't want to consider) and `True` represents "unmasked" pixels (pixels we want to consider). I'll clarify this.

Reply:
AH! I was thinking about masks the wrong way. I thought "masked" data was the data you cared about.